File: two_pointers/1004.java
class Solution {
    public int longestOnes(int[] A, int K) {
        int ret = 0, start = 0, count = 0;
        for (int end = 0; end < A.length; end++) {
            if (A[end] == 0) {
                count++;
            }
            if (count > K) {
                if (A[start] == 0) {
                    count--;
                }
                start++;
            }            
            if (end - start + 1 > ret) {
                ret = end - start + 1;
            }
        }
        return ret;
    }
}

File: sliding_windows/1004.cc
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int curk = 0, left = 0, right = 0, ret = 0;
        while (right < nums.size()) {
            if (nums[right] == 0)
                ++curk;
            if (curk > k) {
                while (nums[left] == 1) {
                    ++left;
                }
                --curk;
                ++left;
            }
            ret = std::max(ret, right-left+1);
            ++right;
        }
        return ret;
    }
};
